src/library/scala/math/BigInt.scala

```diff
@@ -267,18 +442,19 @@ final class BigInt(val bigInteger: BigInteger)
 
   /** Returns a BigInt whose value is the negation of this BigInt
    */
-  def unary_- : BigInt   = new BigInt(this.bigInteger.negate())
+  def unary_- : BigInt   =
+    if (longEncoding) new BigInt(-_long) else new BigInt(this.bigInteger.negate())
```

Comment:
if we are in the cache, `- x` will generally be in the cache too. Should we not check?

Reply:
Same as above; but I won't cherry pick which paths use the caching `.apply` and which paths use a fresh instance, as I don't have a good idea of the performance profile of user code using `BigInt`.